healthz/checks/checks.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package checks
+
+import (
+	"fmt"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/satellite/agent/health"
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+	"github.com/gravitational/satellite/healthz/config"
+	"github.com/gravitational/satellite/monitoring"
+)
+
+// RunAll runs all checks successively and reports general cluster status
+func RunAll(cfg config.Config) (*pb.Probe, error) {
+	var probes health.Probes
+	var checkers health.Checkers
+
+	checkers.AddChecker(monitoring.KubeAPIServerHealth(cfg.KubeAddr))
+	checkers.AddChecker(monitoring.NodesStatusHealth(cfg.KubeAddr))
+	etcdChecker, err := monitoring.EtcdHealth(&cfg.ETCDConfig)
+	if err != nil {
```

Comment:
you don't need to do this every time you run tests if factor checkers configuration to happen outside of RunAll(). 
You can create a type to abstract this away, and have config/run steps - this will also make RunAll not depend on config.